flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -262,9 +265,67 @@ public String toString() {
 			taskInfo.setContainer(containerInfo);
 		}
 
+		containerInfo.addAllVolumes(volumes(params.containerVolumes()));
+
 		return taskInfo.build();
 	}
 
+	/**
+	 * Used to build volume specs for mesos. This allows for mounting additional volumes into a container
+	 *
+	 * @param containerVolumes a comma delimited optional string of [host_path:]container_path[:RO|RW] that
+	 *                         defines mount points for a container volume. If None or empty string, returns
+	 *                         an empty iterator
+	 */
+	public List<Protos.Volume> volumes(Option<String> containerVolumes) {
+		if (containerVolumes.isEmpty()) {
+			return new ArrayList<Protos.Volume>();
+		}
+		String[] specs = containerVolumes.get().split(",");
+		List<Protos.Volume> vols = new ArrayList<Protos.Volume>();
+		for (int i = 0; i < specs.length; i++) {
+			String s = specs[i];
+			if (s.isEmpty()) {
+				continue;
+			}
+			Protos.Volume.Builder vol = Protos.Volume.newBuilder();
+			vol.setMode(Protos.Volume.Mode.RW);
+
+			List<String> parts = Arrays.asList(s.split(":"));
```

Comment:
I thought of a regex first, but after looking at the way the spark framework handles this as well as looking at the regex, it seemed uglier (especially for all paths...). It also doesn't buy us much. 

The biggest problem is that in the case of size == 2, since both a host path and RW/RO is optional, we can't know which one was intended if doesn't explicitly match a `RW` or `RO`.

For example
`/mnt/my_path:RWx`, while it might be logical to assume to that its a typo and what they wanted was a container path of `/mnt/my_path` and `RW` as a mode, it is just as valid for that to be a host path `/mnt/my_path` and a container path relative to the workdir of `RWx`.

The format for this is taken from spark framework, which in turn the format that the docker-cli  run command uses. Ideally, you might have something like a json or yaml description to define these with named keys to remove the ambiguity, however, that would add a lot of complexity, so this still seems reasonable with a slightly sharp edge for the end user